build/net/src/lib.rs

```diff
@@ -18,6 +18,11 @@ pub struct GlobalConfig {
 pub struct NetConfig {
     /// Sockets known to the system, indexed by name.
     pub sockets: BTreeMap<String, SocketConfig>,
+
+    /// VLAN configuration, or None. This is checked against enabled features
+    /// during the `net` build, so it must be present iff the `vlan` feature
+    /// is turned on.
+    pub vlan: Option<VLanConfig>,
```

Comment:
This is an interesting idea!  I tested it out, and have ended up _slightly in favor_ of the current setup.

Making the `vlan` member only exist if the feature is enabled means replacing a bunch of the `Option`-based code in `net/build.rs` with `#[cfg(...)]` blocks.

I'm iffy about this, because it creates code paths that are _conditionally_ compiled. When I tested it, I could only check by building a VLAN-enabled image, then building a no-VLAN image, and often found that I had broken one or the other.  (In general, I see conditional compilation as a necessary evil, but would rather keep all the code building all the time).

It was also more code compared to the `Option`-based version!

Reply:
The down side of this approach is that the vlan code is likely getting included in the netstack text for non-vlan targets. That said, our main focus is the vlan targets right now, so we can fix that if it becomes a problem.

Reply:
This is in the build script 😄 

(and I'm pretty sure the embedded side is doing the right thing)